tests/core/MongoDbOperationStore.spec.ts

```diff
@@ -21,36 +23,36 @@ async function createOperationStore (mongoDbConnectionString: string): Promise<I
  */
 async function createOperationChain (
   createOperation: AnchoredOperationModel,
-  firstUpdateRevealValueEncodedString: string,
   chainLength: number,
-  signingKeyId: string,
+  signingKey: PublicKeyModel,
   signingPrivateKey: JwkEs256k,
   transactionNumber?: number):
   Promise<AnchoredOperationModel[]> {
   const didUniqueSuffix = createOperation.didUniqueSuffix;
   const chain = new Array<AnchoredOperationModel>(createOperation);
-  let updateRevealValueEncodedString = firstUpdateRevealValueEncodedString;
 
+  let currentPublicKey = signingKey;
+  let currentPrivateKey = signingPrivateKey;
   for (let i = 1; i < chainLength ; i++) {
     const transactionNumberToUse = transactionNumber ? transactionNumber : i;
     const transactionTimeToUse = transactionNumberToUse;
 
-    const [nextUpdateRevealValue, nextUpdateCommitmentHash] = OperationGenerator.generateCommitRevealPair();
-    const [newPublicKey] = await OperationGenerator.generateKeyPair(`key${i}`);
+    const [newPublicKey, newPrivateKey] = await OperationGenerator.generateKeyPair(`key${i}`);
     const operationRequest = await OperationGenerator.createUpdateOperationRequestForAddingAKey(
       didUniqueSuffix,
-      updateRevealValueEncodedString,
+      currentPublicKey.jwk,
       newPublicKey,
-      nextUpdateCommitmentHash,
-      signingKeyId,
-      signingPrivateKey
+      Multihash.canonicalizeThenHashThenEncode(newPublicKey.jwk),
```

Comment:
Sanity: no need to pass this parameter as new public key is passed in.

Reply:
We need this because the added key could be another key it's just for testing we pass in the same thing for convenience. (IRL this should never happen because then you are exposing your public key)

Reply:
You are right that the key to add should be separated from the newUpdateKey, I was not thinking clearly. But in which case, i think we should pass in a JWK to this generator method.

Optional: I personally would even prefer to generate and pass an entirely different key, but this would be optional as I understand that adds more burden to the PR.

Reply:
Agree based on the principal that it is fine for test libraries to be more flexible and less usable.